src/image-db.ts

```diff
@@ -102,15 +159,73 @@ class ImageDB {
 
   private createObjectStore(event: IDBVersionChangeEvent) {
     const request: IDBOpenDBRequest = event.target as IDBOpenDBRequest;
+    const transaction: IDBTransaction = request.transaction;
     const db: IDBDatabase = request.result;
 
-    switch (event.oldVersion) {
-      case 1:
+    // Added in version 3, so all paths need to create these stores
```

Comment:
To be honest, I wouldn't care about older versions for now as the app is not public yet.

Reply:
Fair enough. I wrote this upgrade code mostly to teach myself how to do it, and I agree that it isn't needed long term. I've chopped it down to just what is necessary to make the new structure and remove the old one.